catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -110,14 +118,43 @@ public SourceResponse query(QueryRequest request) throws UnsupportedQueryExcepti
 
         SolrQuery query = getSolrQuery(request, filterDelegateFactory.newInstance(resolver));
 
+        boolean isFacetedQuery = false;
+        Serializable textFacetPropRaw = request.getPropertyValue(EXPERIMENTAL_FACET_PROPERTIES_KEY);
+
+        if (textFacetPropRaw != null && textFacetPropRaw instanceof FacetProperties) {
+            FacetProperties textFacetProp = (FacetProperties) textFacetPropRaw;
+            isFacetedQuery = true;
+
+            textFacetProp.getFacetAttributes()
+                    .stream()
+                    .map(this::addAttributeTypeSuffix)
+                    .filter(attr -> attr.contains(String.valueOf(FIRST_CHAR_OF_SUFFIX)))
+                    .forEach(query::addFacetField);
+
+            query.setFacetSort(textFacetProp.getSortKey().name());
+            query.setFacetLimit(textFacetProp.getFacetLimit());
+            query.setFacetMinCount(textFacetProp.getMinFacetCount());
```

Comment:
This value is the minimum number of occurrences a facet value needs in order to be reported.
`query.setFacetMinCount(x)` says "If it doesn't show up at least x times, don't even tell me about it". A max really has no meaning in this context.
The facet limit is the maximum number of different facet values to return.
`query.setFacetLimit(x)` says "Give me the top x values found in this field" which might be what you're thinking here, in which case, we're doing it.